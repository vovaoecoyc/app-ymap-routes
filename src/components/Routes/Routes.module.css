.RoutesBlock {
  display: flex;
  flex-direction: column;
}

.Input {
  width: 200px;
  height: 30px;
  padding: 5px 15px;;
  outline: none;
  border: 1px solid #2779BD;
}