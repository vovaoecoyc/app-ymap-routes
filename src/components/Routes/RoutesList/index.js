import React from 'react';

import RoutesItem from './RoutesItem';
import styles from './RoutesList.module.css';

class RoutesList extends React.Component {
  render() {
    const { points } = this.props;
    return (
      <div className={`${styles.RoutesListBlock}`}>
        {points.map((value, i) => (
          <RoutesItem key={i} id={i} point={value} />
        ))}
      </div>
    );
  }
}

export default RoutesList;
