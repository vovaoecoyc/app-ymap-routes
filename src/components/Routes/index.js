import React from 'react';

// import { Input } from '../UI/Input';
import RoutesList from './RoutesList';
import styles from './Routes.module.css';
import { AppContext } from '../../Context';

class Routes extends React.Component {
  state = {
    value: '',
  };

  static contextType = AppContext;

  handlerChange = e => {
    let value = { value: e.target.value };
    this.setState(() => ({ ...value }));
  };

  handlerKeyPress = e => {
    if (e.key === 'Enter') {
      const mapInstanse = this.context.getMapInstance();
      let newData = localStorage.getItem('points')
          ? JSON.parse(localStorage.getItem('points'))
          : [],
        point = {
          coordinates: [],
          name: this.state.value,
        };
      mapInstanse.load(map => {
        map
          .geocode(this.state.value)
          .then(response => {
            let newPoint =
              response.geoObjects.properties._data.metaDataProperty.GeocoderResponseMetaData.Point;
            if (newPoint) {
              point.coordinates = newPoint.coordinates.reverse();
              newData.push(point);
              localStorage.setItem('points', JSON.stringify(newData));
              this.setState(prevState => ({ ...prevState, value: '' }));
              this.context.loadPoints();
            }
          })
          .catch(error => {
            console.log(error);
          });
      });
    }
  };

  render() {
    let { points } = this.props;
    return (
      <div className={`${styles.RoutesBlock}`}>
        <input
          type="text"
          className={`${styles.Input}`}
          placeholder="Новая точка маршрута"
          value={this.state.value}
          onChange={this.handlerChange}
          onKeyPress={this.handlerKeyPress}
        />
        {/* <Input
          placeholder="Новая точка маршрута"
          onKeyPress={this.handlerKeyPress}
          onChange={this.handlerChange}
          value={this.state.value}
        /> */}
        <RoutesList points={points} />
      </div>
    );
  }
}

export default Routes;
