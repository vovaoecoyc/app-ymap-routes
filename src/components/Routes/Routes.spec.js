import React from 'react';
import { shallow } from 'enzyme';

import Routes from '../Routes';
import context from '../../MockContext';

describe('Routes container', () => {
  const props = {
    points: context.points,
  };

  const initialState = {
    value: '',
  };

  const containerRoutes = shallow(<Routes {...props} />);

  describe('initial render', () => {
    it('RoutesList initial snapshot', () => {
      expect(containerRoutes).toMatchSnapshot();
    });

    it('Is render input', () => {
      expect(containerRoutes.find('input')).toHaveLength(1);
    });

    it('Is render RoutesList', () => {
      expect(containerRoutes.find('RoutesList')).toHaveLength(1);
    });

    it('initial render Routes component', () => {
      containerRoutes.setState(initialState);
      expect(containerRoutes.state()).toEqual(initialState);
    });
  });

  describe('change', () => {
    const valueInput = 'example string';

    beforeEach(() => {
      containerRoutes.find('input').simulate('change', {
        target: {
          value: valueInput,
        },
      });
    });

    it('handler update value of state', () => {
      expect(containerRoutes.state().value).toBe(valueInput);
    });
  });

  /*describe('keypress', () => {
    const keyValue = 'Enter';

    beforeEach(() => {
      containerRoutes.find('input').simulate('keypress', {
        key: keyValue,
      });
    });

    it('handler keypress', () => {
      expect(containerRoutes.state().value).toBe('');
    });
  });*/
});
