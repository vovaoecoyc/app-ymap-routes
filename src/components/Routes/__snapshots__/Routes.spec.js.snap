// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes container initial render RoutesList initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes
    points={
      Array [
        Object {
          "coordinates": Array [
            55.1,
            38.1,
          ],
          "name": "Name Point1",
        },
        Object {
          "coordinates": Array [
            55.9,
            38,
          ],
          "name": "Name Point2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="Input"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Новая точка маршрута"
          type="text"
          value=""
        />,
        <RoutesList
          points={
            Array [
              Object {
                "coordinates": Array [
                  55.1,
                  38.1,
                ],
                "name": "Name Point1",
              },
              Object {
                "coordinates": Array [
                  55.9,
                  38,
                ],
                "name": "Name Point2",
              },
            ]
          }
        />,
      ],
      "className": "RoutesBlock",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "Input",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "Новая точка маршрута",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "points": Array [
            Object {
              "coordinates": Array [
                55.1,
                38.1,
              ],
              "name": "Name Point1",
            },
            Object {
              "coordinates": Array [
                55.9,
                38,
              ],
              "name": "Name Point2",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="Input"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Новая точка маршрута"
            type="text"
            value=""
          />,
          <RoutesList
            points={
              Array [
                Object {
                  "coordinates": Array [
                    55.1,
                    38.1,
                  ],
                  "name": "Name Point1",
                },
                Object {
                  "coordinates": Array [
                    55.9,
                    38,
                  ],
                  "name": "Name Point2",
                },
              ]
            }
          />,
        ],
        "className": "RoutesBlock",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Input",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Новая точка маршрута",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "points": Array [
              Object {
                "coordinates": Array [
                  55.1,
                  38.1,
                ],
                "name": "Name Point1",
              },
              Object {
                "coordinates": Array [
                  55.9,
                  38,
                ],
                "name": "Name Point2",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
